node.go

```diff
@@ -1440,18 +1440,20 @@ func (r *nodeRegistry) get(uid string) (*controlpb.Node, bool) {
 func (r *nodeRegistry) add(info *controlpb.Node) bool {
 	var isNewNode bool
 	r.mu.Lock()
-	if node, ok := r.nodes[info.Uid]; ok {
+	if _, ok := r.nodes[info.Uid]; ok {
 		if info.Metrics != nil {
 			r.nodes[info.Uid] = info
 		} else {
-			node.Version = info.Version
-			node.NumChannels = info.NumChannels
-			node.NumClients = info.NumClients
-			node.NumUsers = info.NumUsers
-			node.NumSubs = info.NumSubs
-			node.Uptime = info.Uptime
-			node.Data = info.Data
-			r.nodes[info.Uid] = node
+			r.nodes[info.Uid] = &controlpb.Node{
```

Comment:
Think with this approach we also need to set `controlpb.Node.Name` field - previously it was not updated since supposed to be stable for existing node with certain UID in registry. Also, `Metrics` is becoming nil here while we need to keep previous value. 

Why this change btw - it seems to do mostly the same as before?

Reply:
`Node.Info()` gets the `*controlpb.Node` pointer by calling `nodeRegistery.list()`, and reads some fields of `*controlpb.Node`, but in the same time, some  `*controlpb.Node`attributes writing may happend here, so this causes this data race error:
```
==================
WARNING: DATA RACE
Write at 0x00c001358678 by goroutine 2940:
  github.com/centrifugal/centrifuge.(*nodeRegistry).add()
      /home/runner/work/centrifuge/centrifuge/node.go:1447 +0x23d
  github.com/centrifugal/centrifuge.(*Node).nodeCmd()
      /home/runner/work/centrifuge/centrifuge/node.go:1071 +0x65
  github.com/centrifugal/centrifuge.(*Node).pubNode()
      /home/runner/work/centrifuge/centrifuge/node.go:907 +0x892
  github.com/centrifugal/centrifuge.(*Node).nodeCmd()
      /home/runner/work/centrifuge/centrifuge/node.go:1074 +0x10f
  github.com/centrifugal/centrifuge.(*Node).handleControl()
      /home/runner/work/centrifuge/centrifuge/node.go:653 +0x992
  github.com/centrifugal/centrifuge.(*brokerEventHandler).HandleControl()
      /home/runner/work/centrifuge/centrifuge/node.go:1568 +0x6e
  github.com/centrifugal/centrifuge.(*RedisBroker).runControlPubSub.func4()
      /home/runner/work/centrifuge/centrifuge/broker_redis.go:916 +0x216

Previous read at 0x00c001358678 by goroutine 1866:
  github.com/centrifugal/centrifuge.(*Node).Info()
      /home/runner/work/centrifuge/centrifuge/node.go:604 +0x172
  github.com/centrifugal/centrifuge.waitAllNodes()
      /home/runner/work/centrifuge/centrifuge/broker_redis_test.go:893 +0x66
  github.com/centrifugal/centrifuge.BenchmarkRedisSurvey.func1()
      /home/runner/work/centrifuge/centrifuge/broker_redis_test.go:978 +0xb64
  testing.(*B).runN()
      /opt/hostedtoolcache/go/1.17.12/x64/src/testing/benchmark.go:192 +0x1e1
  testing.(*B).run1.func1()
      /opt/hostedtoolcache/go/1.17.12/x64/src/testing/benchmark.go:232 +0x78

Goroutine 2940 (running) created at:
  github.com/centrifugal/centrifuge.(*RedisBroker).runControlPubSub()
      /home/runner/work/centrifuge/centrifuge/broker_redis.go:906 +0x684
  github.com/centrifugal/centrifuge.(*RedisBroker).runShard.func4()
      /home/runner/work/centrifuge/centrifuge/broker_redis.go:375 +0x6e
  github.com/centrifugal/centrifuge.(*RedisBroker).runForever()
      /home/runner/work/centrifuge/centrifuge/broker_redis.go:338 +0xe2
  github.com/centrifugal/centrifuge.(*RedisBroker).runShard·dwrap·10()
      /home/runner/work/centrifuge/centrifuge/broker_redis.go:374 +0x47

Goroutine 1866 (running) created at:
  testing.(*B).run1()
      /opt/hostedtoolcache/go/1.17.12/x64/src/testing/benchmark.go:225 +0x173
  testing.(*B).Run()
      /opt/hostedtoolcache/go/1.17.12/x64/src/testing/benchmark.go:670 +0x82b
  github.com/centrifugal/centrifuge.BenchmarkRedisSurvey()
      /home/runner/work/centrifuge/centrifuge/broker_redis_test.go:[929](https://github.com/j178/centrifuge/runs/7702777745?check_suite_focus=true#step:5:930) +0x18f
  testing.(*B).runN()
      /opt/hostedtoolcache/go/1.17.12/x64/src/testing/benchmark.go:192 +0x1e1
  testing.(*B).run1.func1()
      /opt/hostedtoolcache/go/1.17.12/x64/src/testing/benchmark.go:232 +0x78
==================
```

This approach creates new `controlpb.Node` object, instead of overwriting the original shared one, to avoid this data race.

Reply:
I see, yep - makes sense, thank you!